Build failure: nickel

### Steps To Reproduce

Steps to reproduce the behavior:

1. build `nickel` on unstable on `aarch64-darwin`


### Build log

```
error: could not compile `nickel-lang-cli` (bin "nickel") due to previous error
warning: build failed, waiting for other jobs to finish...
error: linking with `/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin/cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/nix/store/2jdssgjk1i4gjz4w9k8nz3831wq7sxdw-rustc-1.73.0/lib/rustlib/aarch64-apple-darwin/bin:/nix/store/46zkgqp0wk7rwd2n7paa561ch5xjkkyz-cargo-1.73.0/bin:/>
  = note: Undefined symbols for architecture arm64:
            "___gxx_personality_v0", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
            "___cxa_allocate_exception", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
            "___cxa_free_exception", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
            "vtable for std::length_error", referenced from:
                std::length_error::length_error[abi:v160006](char const*) in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
            NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
            "std::length_error::~length_error()", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
            "typeinfo for std::length_error", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
            "operator delete(void*)", referenced from:
                _tree_sitter_nickel_external_scanner_scan in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
                _tree_sitter_nickel_external_scanner_deserialize in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
                _tree_sitter_nickel_external_scanner_destroy in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
            "___cxa_throw", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
            "operator new(unsigned long)", referenced from:
                _tree_sitter_nickel_external_scanner_create in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
                _tree_sitter_nickel_external_scanner_scan in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
                _tree_sitter_nickel_external_scanner_deserialize in libtree_sitter_nickel-bd2d14b79075b75f.rlib(scanner.o)
          ld: symbol(s) not found for architecture arm64
          clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: could not compile `nickel-lang-lsp` (bin "nls") due to previous error
```

### Additional context

Nickel was updated to 1.3.0 in #267932 

### Notify maintainers

@AndersonTorres @felschr @matthiasbeyer + @yannham (PR author)
### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
[user@system:~]$ nix run nixpkgs#nix-info -- "-m"                                                                                                
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.0`
 - nixpkgs: `/nix/store/00haf0fr1410xvgcqpcy6f5dahaqcv28-source`
```

--- a/pkgs/by-name/ni/nickel/package.nix
+++ b/pkgs/by-name/ni/nickel/package.nix
@@ -1,6 +1,7 @@
 { lib
 , rustPlatform
 , fetchFromGitHub
+, stdenv
 , python3
 , nix-update-script
 }:
@@ -35,6 +36,11 @@ rustPlatform.buildRustPackage rec {
   };
 
   cargoBuildFlags = [ "-p nickel-lang-cli" "-p nickel-lang-lsp" ];
+
+  env = lib.optionalAttrs stdenv.cc.isClang {
+    # Work around https://github.com/NixOS/nixpkgs/issues/166205.
+    NIX_LDFLAGS = "-l${stdenv.cc.libcxx.cxxabi.libName}";
+  };
 
   nativeBuildInputs = [
     python3

